galley/pkg/crd/validation/webhook.go

```diff
@@ -219,10 +296,26 @@ func (wh *Webhook) Run(stop <-chan struct{}) {
 	}()
 	defer wh.stop()
 
+	log.Infof("Checking if %s/%s is ready before registering webhook configuration ",
+		wh.deploymentNamespace, wh.deploymentName)
+	if shutdown := wh.waitForEndpointReady(stop); shutdown {
+		log.Info("Endpoint readiness check stopped - controller shutting down")
```

Comment:
This can be an error not just shutdown, like k8s controllers logic.

Reply:
When would this be an error? AFAICT, cache.WaitForCacheSync() only returns false if stop was requested, i.e. the stop channel was closed due to shutdown.

Reply:
Yup